src/Altinn.App.Core/Internal/Pdf/IPdfService.cs

```diff
@@ -22,5 +22,6 @@ public interface IPdfService
     /// <param name="instance">The instance details.</param>
     /// <param name="taskId">The task id for witch the pdf is generated</param>
     /// <param name="ct">Cancellation Token for when a request should be stopped before it's completed.</param>
-    Task<Stream> GeneratePdf(Instance instance, string taskId, CancellationToken ct);
+    /// <param name="isPreview">If set to true, this will show a text saying that this is a preview PDF.</param>
+    Task<Stream> GeneratePdf(Instance instance, string taskId, CancellationToken ct, Boolean? isPreview = false);
```

Comment:
CancellationToken conventionally comes last. Not sure what is most important. Keeping existing calls working, or following conventions.

The middle ground would be to have two methods (something like this)
```suggestion
    [Obsolete("Use the GeneratePdf method with the isPreview parameter")]
    Task<Stream> GeneratePdf(Instance instance, string taskId, CancellationToken ct);
    
    Task<Stream> GeneratePdf(Instance instance, string taskId, bool isPreview, CancellationToken ct);
```

At least use `bool` instead of `Boolean?` when `null` doesn't mean anything.

Reply:
For the implementation, just use a stub implementation for the obsolete method

```c#
    Task<Stream> GeneratePdf(Instance instance, string taskId, CancellationToken ct)
    {
        await GeneratePdf(instance, taskId, isPreview:false, ct);
    }
```

Reply:
Well according to the spec its fine when you have optional parameters:

https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1068#special-cases

I would just go with the optional parameter, but if you guys prefer two methods I dont mind :)

Reply:
We decided to have 2 methods like you propose, but not obsoleting the original method as it is fine as is